compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -101,15 +112,16 @@ public PersistRecordValidator() {
         isPersistEntity = false;
         tableName = "";
         noOfReportDiagnostic = 0;
-        referenceTables = new HashMap<>();
         tableNames = new ArrayList<>();
         recordNames = new ArrayList<>();
-        tableNamesInScript = new ArrayList<>();
         isNewBuild = true;
     }
 
     @Override
     public void perform(SyntaxNodeAnalysisContext ctx) {
+        if (!(ctx.node() instanceof TypeDefinitionNode)) {
+            return;
+        }
```

Comment:
Is there a possibility for this? This task is only applied for TypeDefinition nodes.

Reply:
Removed in [8e45254](https://github.com/ballerina-platform/module-ballerina-persist/pull/67/commits/8e452547f11586ee50ea78271d144703ae4f75c3)